src/e2e-tests/page-objects/ChangePreferredStatusGloballyPopup.ts

```diff
@@ -0,0 +1,54 @@
+// SPDX-FileCopyrightText: Meta Platforms, Inc. and its affiliates
+// SPDX-FileCopyrightText: TNG Technology Consulting GmbH <https://www.tngtech.com>
+//
+// SPDX-License-Identifier: Apache-2.0
+import { expect, type Locator, type Page } from '@playwright/test';
+
+import { text } from '../../shared/text';
+
+export class ChangePreferredStatusGloballyPopup {
+  private readonly window: Page;
+  private readonly node: Locator;
+  readonly okButton: Locator;
+  readonly undoButton: Locator;
+
+  constructor(window: Page) {
+    this.window = window;
+    this.node = window.getByLabel('change preferred status globally popup');
+    this.okButton = this.node.getByRole('button', {
+      name: 'Ok',
+      exact: true,
+    });
+    this.undoButton = this.node.getByRole('button', {
+      name: 'Undo',
+      exact: true,
+    });
+  }
+
+  public assert = {
+    isVisible: async (): Promise<void> => {
+      await expect(this.node).toBeVisible();
+    },
+    isHidden: async (): Promise<void> => {
+      await expect(this.node).toBeHidden();
+    },
+    showsMarkAsPreferredWarning: async (): Promise<void> => {
```

Comment:
let's stick with the pattern:
```suggestion
    markAsPreferredWarningIsVisible: async (): Promise<void> => {
```
and similar for the hidden state.